### ISSUE TITLE
etoc package incompatible

### ISSUE THREAD

**Issue by mbertucci47** [latex3/tagging-project#598]
The [etoc](https://ctan.org/pkg/etoc) package produces parent-child warnings with the tagging code. Example:
```tex
\DocumentMetadata
  {
    lang=en-US,
    pdfversion=2.0,
    pdfstandard=ua-2,
    testphase={phase-III,math,title,table,firstaid}
  }
\documentclass{article}

\usepackage{etoc}

\begin{document}

\tableofcontents

\section{blub}
\subsection{blub blub}

\end{document}
```

**Comment by jfbu** [comment]
FYI, after updating on Sep. 18, 2025 the [etoc tagging branch](https://github.com/jfbu/etoc/tree/tagging) (which was previously last updated on Jan. 15, 2024) to inject `\UseTaggingSocket{toc/starttoc/before}{toc}` in place of former `\@starttoc@cfgpoint@before{toc}`, the MWE appears to compile successfully, as far as I can tell with testing on ngpdf.com.  It probably did in January 2024 with the `etoc` tagging branch and the then in place LaTeX2e. 

I [summarized](https://github.com/jfbu/etoc/blob/tagging/etoctagging.md) in January 2024 my conclusions of that time which were that it looked impossible to support tagging in documents where users exploit to the fullest the `etoc` capabilities.  But many people use it only for its `\localtableofcontents`.  If I get time and it looks feasible to fully support tagging in documents using only `\localtableofcontents` in "compatibility mode" (i.e. `\contentsline` is unmodified), I expect this to be part of `etoc 1.2e` release.

**Comment by matteofg** [comment]
Does this mean the package is now compatible with tagging?
If so, is it also compatible with the new heading template mechanism?

**Comment by mbertucci47** [comment]
@matteofg No, the example above still produces warnings and several other package functions would need significant changes.
